src/main/java/org/openstreetmap/atlas/tags/annotations/extraction/LayerExtractor.java

```diff
@@ -0,0 +1,57 @@
+package org.openstreetmap.atlas.tags.annotations.extraction;
+
+import java.util.Optional;
+
+import org.openstreetmap.atlas.tags.BridgeTag;
+import org.openstreetmap.atlas.tags.LayerTag;
+import org.openstreetmap.atlas.tags.Taggable;
+import org.openstreetmap.atlas.tags.TunnelTag;
+
+/**
+ * Extracts layer values.
+ *
+ * @author bbreithaupt
+ */
+public final class LayerExtractor
+{
+    /**
+     * Extracts layer values as {@link Optional} {@link Long}s based on the {@link LayerTag},
+     * {@link BridgeTag}, and {@link TunnelTag}s. Defaults to 0 if no layer value is found.
+     *
+     * @param taggable
+     *            {@link Taggable} to get a value for
+     * @return {@link Optional} {@link Long} layer value. Will return empty if taggable has a
+     *         {@link LayerTag} that is invalid.
+     */
+    public static Optional<Long> validateAndExtract(final Taggable taggable)
```

Comment:
There is some overlap between this functionality and the implementation in `LayerTag`. It might be worth keeping the tests and adding some of the bridge and tunnel logic to the existing extraction logic.

Reply:
I thought about making this an addition to that logic, but was not sure about adding logic to look at other tags from inside a tag class. I am not sure that there is any other tag class where we do that. However, if we think that is fine then I am happy to handle this all from the LayerTag class.

Reply:
Good callout, my vote is that this is a normal usage pattern. An example of this is in `HighwayTag`, which references `AreaTag`.

Reply:
All the logic has been consolidated into `LayerTag.getTaggedOrImpliedValue()`.